test/utils/infer.py

```diff
@@ -42,7 +42,9 @@ def infer_with_circle_interpreter(
         The output produced by the 'circle-interpreter'
     """
     circle_model = tico.utils.model.CircleModel.load(circle_path)
-    circle_result = circle_model(*forward_args, **forward_kwargs)
+    ispec = ModelInputSpec.load(circle_path)
+    inputs = ispec.bind(forward_args, forward_kwargs, check=True)
+    circle_result = circle_model(*inputs)
```

Comment:
Introducing feature looks very good. But I guess it would be more helpful if input bindings are automatically done inside the CircleModel like below. 

```python
circle_model = tico.utils.model.CircleModel.load(circle_path)
circle_result = circle_model(*forward_args, **forward_kwargs) # input bindings are done inside the `__call__` method.
```

Anyway, my suggestion doesn't have to be applied in this PR. Let's consider this later.